dali/python/nvidia/dali/plugin/numba/experimental/__init__.py

```diff
@@ -549,17 +550,30 @@ def _check_minimal_numba_version(throw: bool = True):
 
     @staticmethod
     def _check_cuda_compatibility(throw: bool = True):
-        toolkit_version = cuda.runtime.get_version()
-        driver_version = cuda.driver.driver.get_version()
+        toolkit_version = nb_cuda.runtime.get_version()
+        driver_version = nb_cuda.driver.driver.get_version()
 
         # numba_cuda should handle the compatibility between toolkit and driver versions
         # otherwise check if if driver and runtime matches, or if the last working numba version
         # matches the driver for CUDA 12
-        if not importlib.util.find_spec("numba_cuda") and (
+        try:
+            # try importing cuda.core as it can be used later to check the compatibility
+            # if is okay to fail as it may not be installed, the check later can handle this
+            import cuda.core
+        except ImportError:
+            pass
+        if (
+            not importlib.util.find_spec("numba_cuda")
+            or (
+                importlib.util.find_spec("cuda.core")
+                and Version(cuda.core.__version__) <= Version("0.31.1")
```

Comment:
I think you mixed the cuda.core and numba versioning, latest cuda.core is 0.3.1:
<img width="252" height="276" alt="obraz" src="https://github.com/user-attachments/assets/47e56f8b-8fa2-480f-85c4-255ec6733a92" />

Reply:
My bad. Fixed